### ISSUE TITLE
untracked pygoturn dependency in carla baseline models

### ISSUE THREAD

**Issue by mwartell** [twosixlabs/armory#1302]
In [armory/baseline_models/pytorch/carla_goturn.py](https://github.com/twosixlabs/armory/blob/2b8dfe5925cb2563ce899e68f2555c129b00d410/armory/baseline_models/pytorch/carla_goturn.py#L6-L9) there is an import from a module which is required to be present but isn't in our source tree.

The problem is that pygoturn was never released on github nor on pypi so neither

    pip install pygoturn
    pip install git+https://github.com/amoudgl/pygoturn.git@1785ca7106fc0aa9e0fd03b440f34454e8a10e78 

work. 

The source repository hasn't had a commit since 2019, so I consider it effectively closed. I think the best approach is to fork the repo, add a pyproject.toml thus making the forked repo installable.

**Comment by mwartell** [comment]
I've started work in https://github.com/mwartell/pygoturn. If this works it should probably be re-homed the the armory github account so that it isn't tied to me.

**Comment by davidslater** [comment]
Currently, we treat this as an "external" github repo in our config:
https://github.com/twosixlabs/armory/blob/master/scenario_configs/carla_video_tracking_goturn_advtextures_defended.json#L60

Similar to how we use the MARS model in Ucf101:
https://github.com/twosixlabs/armory/blob/master/armory/baseline_models/pytorch/ucf101_mars.py#L14-L16

SincNet:
https://github.com/twosixlabs/armory/blob/master/armory/baseline_models/pytorch/sincnet.py#L19

DeepSpeech is even worse (I think).

**Comment by davidslater** [comment]
I think that we should treat these in a consistent way across the repo.

Maybe we restructure `baseline_models` and have a `baseline_models/external` folder that contains all of these (with some clear notation at the top, and possibly even directly handling the armory download process.

Then the remaining models can be ones that run with one of our base containers.

**Comment by mwartell** [comment]
I'm now less certain that anything needs to be done here. I think the broader question I left unstated was "why are we dynamically loading a library that hasn't changed since 2019?". What I saw as an escape-hatch of developer convenience is actually forced by the application needs. Neither MARS nor SincNet have releases so the github repo is really the only source for those modules.

I think your `baseline_models/external` suggestion means that libraries like pygoturn we could snapshot the repo in `external` and have that available for `import`. If we bother to do this, I expect there would be some cases where a snapshot isn't current enough and that the demand load from github will have to remain.

**Comment by davidslater** [comment]
No, I wasn't suggesting that we keep snapshots in the repo - I was suggesting that we make it clear that the set of files in `baseline_models/external` requires additional external dependencies that are typically pulled via git and modifying those files to explicitly describe what is required and make it easy to pull in via the built-in armory github pull mechanism.
